### ISSUE TITLE
Inspec Compliance upload fails via cli but succeeds using the UI

### ISSUE THREAD

**Issue by paul1994** [inspec/inspec#4498]
### 🎛 Description

**Choose one:** is this a 🐛 bug report 
When trying to upload a profile to automate we receive an error when trying with the cli. The profile should be able to upload to the automate server via the cli.
Note that with the same profile I could upload to automate with the UI

### 🌍 InSpec and Platform Version

windows 10

Tell us which version of InSpec (`inspec version` or SHA of a commit) and Operating System distribution you are using

### 🤔 Replication Case

Created Inspec profile`inpsec init profile test-profile`, logged into Automate`inspec compliance login https://automate.<domain_name> --user admin --insecure --token '<security-token>'', archived the profile`inspec archive "path_to_profile`, all previous steps worked without issue.
Issue trying to upload profile to Automate`inspec compliance upload .\test-profile-0.1.1.tar.gz'
ERROR:

(reformatted by @zenspider):
```
inpsec init profile test-profile
inspec compliance login https://automate.<domain_name> --user admin --insecure --token "<security-token>''
inspec archive path/to/profile
inspec compliance upload .\test-profile-0.1.1.tar.gz
ERROR:

Traceback (most recent call last):
........
C:/opscode/.../inspec-4.7.3/lib/inspec/metadata.rb:67:in `supports_platform?':
undefined method `platform' for
#<#<Class:0x0000000005b82a68>:0x0000000006a644c8>
(NoMethodError)
```

### 💁 Possible Solutions

If you have already ideas how to solve the issue, add them here.

### 💻 Stacktrace

```
PS C:\Users\admin\ inspec compliance upload test-profile

Dependencies for profile test-profile successfully vendored to C:\Users\admin\test-profile\vendor

I, [2019-09-17T16:10:30.599422 #7340]  INFO -- : Checking profile in test-profile

I, [2019-09-17T16:10:30.600023 #7340]  INFO -- : Metadata OK.

Traceback (most recent call last):

        26: from C:/opscode/chef-workstation/bin/inspec:359:in `<main>'
        25: from C:/opscode/chef-workstation/bin/inspec:359:in `load'
        24: from $GEMS/inspec-bin-4.7.3/bin/inspec:11:in `<top (required)>'
        23: from $GEMS/inspec-4.7.3/lib/inspec/base_cli.rb:31:in `start'
        22: from $GEMS/thor-0.20.3/lib/thor/base.rb:466:in `start'
        21: from $GEMS/thor-0.20.3/lib/thor.rb:387:in `dispatch'
        20: from $GEMS/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
        19: from $GEMS/thor-0.20.3/lib/thor/command.rb:27:in `run'
        18: from $GEMS/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
        17: from $GEMS/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
        16: from $GEMS/thor-0.20.3/lib/thor.rb:387:in `dispatch'
        15: from $GEMS/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
        14: from $GEMS/thor-0.20.3/lib/thor/command.rb:27:in `run'
        13: from $GEMS/inspec-4.7.3/lib/plugins/inspec-compliance/lib/inspec-compliance/cli.rb:160:in `upload'
        12: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:406:in `check'
        11: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:435:in `controls_count'
        10: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:184:in `params'
         9: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:581:in `load_params'
         8: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:588:in `load_checks_params'
         7: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:192:in `collect_tests'
         6: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:192:in `each'
         5: from $GEMS/inspec-4.7.3/lib/inspec/profile.rb:195:in `block in collect_tests'
         4: from $GEMS/inspec-4.7.3/lib/inspec/profile_context.rb:144:in `load_control_file'
         3: from $GEMS/inspec-4.7.3/lib/inspec/profile_context.rb:160:in `load_with_context'
         2: from $GEMS/inspec-4.7.3/lib/inspec/profile_context.rb:160:in `instance_eval'
         1: from test-profile/controls/example.rb:6:in `load_with_context'

C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/inspec-4.7.3/lib/inspec/dsl.rb:47:in `method_missing': undefined method `file' for #<#<Class:0x0000000005cdbb58>:0x0000000005cd9c68> (NoMethodError)

Did you mean?  fail
```

**Comment by zenspider** [comment]
Other interesting tidbit from the thread, this works fine via automate UI but fails using CLI.

I'm not sure how relevant the windows side of this is. My guess is its not, but it might be some winrm transport + `file` issue for all I know right now.

**Comment by clintoncwolfe** [comment]
#4489 is a duplicate or similar aspect of this issue.

**Comment by clintoncwolfe** [comment]
Based on information provided on #4489, I was able to reproduce the issue on inspec 4.16.0 but not on current tip 4.17.3 . Bisection narrowed it to being fixed on the 4.16.8 nonreleased build, which included the merge of #4365 .

So... our status at this point is that we already have a fix in place, and will be present on the next release of inspec.

If you consume omnibus packages of inspec, note that operating system packages are available for every build - switch your (e.g.) rpm repos to pull from the "current" rather than "stable" channels. Otherwise, we expect to release a normal release on Thursday Sept 26th.

**Comment by clintoncwolfe** [comment]
This is verified fixed on Chef InSpec v4.17.11, released Friday Sept 27 - thanks!
